dev/com.ibm.ws.webcontainer.security/test/com/ibm/ws/webcontainer/security/InMemoryHttpSessionCacheTest.java

```diff
@@ -0,0 +1,208 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.webcontainer.security;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ *
+ */
+public class InMemoryHttpSessionCacheTest {
```

Comment:
Depending on what we decide as far as using `request.getSession(false)` goes, we might need to add unit tests to verify behavior for when an HTTP session ID is null.

Reply:
i added a check to see if an httpsessionid exists for now before inserting. still need to investigate a way to use some other sort of id when an http session is not available.